### PR TITLE
Server-side UI

### PR DESCRIPTION
**Pull request by m-reza-rahman**
Initial working version. Still need to do more cleanup/simplification of the code, more testing, adding server-side validation and more documentation. JUnit tests can be added after the 2.0 release.

### CODE REVIEW

**Comment by m-reza-rahman** [comment]
Hi @jeyvison, you could begin reviewing this. It's days away from basically being ready for the 2.0 release.

File: ui/src/main/java/org/eclipse/starter/ui/Project.java [hunk 1]
```diff
@@ -1,104 +1,191 @@
...
+		return shuffledRuntimes;
+	}
+
+	public void generate() {
+		LOGGER.log(Level.INFO,
+				"Generating project - Jakarta EE version: {0}, Jakarta EE profile: {1}, Java SE version: {2}, Docker: {3}, runtime: {4}",
+				new Object[] { jakartaVersion, profile, javaVersion, docker, runtime });
+
+		try {
+			FacesContext facesContext = FacesContext.getCurrentInstance();
```
Comment (hantsy) [inline ui/src/main/java/org/eclipse/starter/ui/Project.java:99]: FacesContext, ExternalContext can be injected directly.

File: ui/src/main/java/org/eclipse/starter/ui/Project.java [hunk 2]
```diff
@@ -1,104 +1,191 @@
...
+				"-Druntime=" + runtime };
+		int result = new MavenCli().doMain(options, workingDirectory.getAbsolutePath(), System.out, System.err);
+
+		if (result != 0) {
+			throw new RuntimeException("Failed to invoke Maven Archetype.");
+		}
+	}
+
+	private void generateZip(File workingDirectory) {
+		try {
```
Comment (hantsy) [inline ui/src/main/java/org/eclipse/starter/ui/Project.java:134]: Use try/resources for the `AutoCloseable` APIs.

File: ui/src/main/java/org/eclipse/starter/ui/Project.java (hunk 1 above)
Comment (m-reza-rahman) [reply ui/src/main/java/org/eclipse/starter/ui/Project.java]: Addressed.

File: ui/src/main/java/org/eclipse/starter/ui/Project.java (hunk 2 above)
Comment (m-reza-rahman) [reply ui/src/main/java/org/eclipse/starter/ui/Project.java]: Addressed.

**Comment by m-reza-rahman** [comment]
I think it’s good enough for an intermediate merge now, @jeyvison. Marking it as such. Will continue to work on it this week.

**Pull request merged**